tools/tsp-client/src/index.ts

```diff
@@ -243,111 +382,236 @@ async function convert(readme: string, outputDir: string, arm?: boolean): Promis
-          await generate({ rootUrl: outputDir, noCleanup: options.noCleanup, additionalEmitterOptions: options.emitterOptions});
-        }
-        break;
-      case "sync":
-        await syncTspFiles(rootUrl, options.localSpecRepo);
-        break;
-      case "generate":
-        await generate({ rootUrl, noCleanup: options.noCleanup, additionalEmitterOptions: options.emitterOptions});
-        break;
-      case "update":
-        if (options.repo && !options.commit) {
-            throw new Error("Commit SHA is required when specifying `--repo`, please specify a commit using `--commit`");
-        }
-        if (options.commit) {
-          const tspLocation: TspLocation = await readTspLocation(rootUrl);
-          tspLocation.commit = options.commit ?? tspLocation.commit;
-          tspLocation.repo = options.repo ?? tspLocation.repo;
-          await writeFile(joinPaths(rootUrl, "tsp-location.yaml"), `directory: ${tspLocation.directory}\ncommit: ${tspLocation.commit}\nrepo: ${tspLocation.repo}\nadditionalDirectories: ${tspLocation.additionalDirectories}`);
-        } else if (options.tspConfig) {
-          const tspLocation: TspLocation = await readTspLocation(rootUrl);
-          const tspConfig = resolveTspConfigUrl(options.tspConfig);
-          tspLocation.commit = tspConfig.commit ?? tspLocation.commit;
-          tspLocation.repo = tspConfig.repo ?? tspLocation.repo;
-          await writeFile(joinPaths(rootUrl, "tsp-location.yaml"), `directory: ${tspLocation.directory}\ncommit: ${tspLocation.commit}\nrepo: ${tspLocation.repo}\nadditionalDirectories: ${tspLocation.additionalDirectories}`);
-        }
-        await syncTspFiles(rootUrl, options.localSpecRepo);
-        await generate({ rootUrl, noCleanup: options.noCleanup, additionalEmitterOptions: options.emitterOptions});
-        break;
-      case "convert":
-        Logger.info("Converting swagger to typespec...");
-        let readme = options.swaggerReadme!;
-        if (await doesFileExist(readme)) {
-          readme = normalizePath(resolve(readme));
-        }
-        await convert(readme, rootUrl, options.arm);
-        if (options.arm) {
-          try {
-            await unlink(joinPaths(rootUrl, "resources.json"));
-          } catch (err) {
-            Logger.error(`Error occurred while attempting to delete resources.json: ${err}`);
-            process.exit(1);
-          }
-        }
-        break;
-      default:
-        throw new Error(`Unknown command: ${options.command}`);
+  if (useOutputDir.toLowerCase() === "n") {
+    const newOutputDir = prompt("Enter output directory: ");
+    if (!newOutputDir) {
+      Logger.error("Output directory is required");
+      process.exit(1);
+    }
+    outputDir = resolvePath(normalizePath(newOutputDir));
   }
+  Logger.info(`Using output directory '${outputDir}'`);
+  return outputDir;
 }
 
-main().catch((err) => {
+const parser = yargs(hideBin(process.argv))
```

Comment:
I'm not seeing the --version flag anymore, was it removed?

Reply:
It is automatically supported by the framework. 
![image](https://github.com/user-attachments/assets/01514b3d-2a45-40d8-874f-196b3cd60752)